src/Microsoft.OData.Edm/Csdl/CsdlWriter.cs

```diff
@@ -117,6 +178,15 @@ protected virtual void WriteCsdl()
             // nothing here
         }
 
+        /// <summary>
+        /// Asynchronously Writes CSDL output.
+        /// </summary>
+        /// <returns>Task represents an asynchronous operation that may or may not return a result.</returns>
+        protected virtual Task WriteCsdlAsync()
+        {
+            return Task.FromResult(0);

```

Comment:
Why `Task.FromResult` and not `Task.CompletedTask`, this seems to occur in multiple places in the PR. Could you check all usage of this pattern in the PR and confirm whether `Task.CompletedTask` could be used instead?